src/main/java/duke/Command.java

```diff
@@ -0,0 +1,183 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDos;
+import java.io.IOException;
+
+/**
+ * Abstraction of all commands possibly received on duke,
+ * and executes all commands.
+ * @author Shaune Ang
+ */
+public class Command {
+    static final String EXITWORD = "bye";
+    private ToDoList toDoList;
+    private Ui ui;
+    private FileLoaderSaver storage;
+    private String fullCommand;
+
+    private CommandTypes commandType;
+
+    // enum of all command types
+    public enum CommandTypes {
+        LIST,
+        TODO,
+        DEADLINE,
+        EVENT,
+        MARK,
+        UNMARK,
+        DELETE,
+        EXIT
+    }
+
+    /**
+     * Constructor for command objects
+     * @param commandType type of command this command with execute
+     * @param fullCommand full user input command in String
```

Comment:
Following the Javadoc comments convention would be better (empty line between description an parameter section)